pytorch_lightning/callbacks/progress.py

```diff
@@ -382,6 +383,52 @@ def init_test_tqdm(self) -> tqdm:
         )
         return bar
 
+    def get_progress_bar_dict(self, pl_module: "pl.LightningModule") -> Dict[str, Union[int, str]]:
+        r"""
+        Implement this to override the default items displayed in the progress bar.
+        By default it includes the average loss value, split index of BPTT (if used)
+        and the version of the experiment when using a logger.
+
+        .. code-block::
+
+            Epoch 1:   4%|▎         | 40/1095 [00:03<01:37, 10.84it/s, loss=4.501, v_num=10]
+
+        Here is an example how to override the defaults:
+
+        .. code-block:: python
+
+            def get_progress_bar_dict(self, model):
+                # don't show the version number
+                items = super().get_progress_bar_dict(model)
+                items.pop("v_num", None)
+                return items
+
+        Return:
+            Dictionary with the items to be displayed in the progress bar.
+        """
+        # call .item() only once but store elements without graphs
+        running_train_loss = self.trainer.fit_loop.running_loss.mean()
+        avg_training_loss = None
+        if running_train_loss is not None:
+            avg_training_loss = running_train_loss.cpu().item()
+        elif pl_module.automatic_optimization:
+            avg_training_loss = float("NaN")
+
+        tqdm_dict = {}
+        if avg_training_loss is not None:
+            tqdm_dict["loss"] = f"{avg_training_loss:.3g}"
+
+        if pl_module.truncated_bptt_steps > 0:
+            tqdm_dict["split_idx"] = self.trainer.fit_loop.split_idx
+
+        if self.trainer.logger is not None and self.trainer.logger.version is not None:
+            version = self.trainer.logger.version
+            # show last 4 places of long version strings
+            version = version[-4:] if isinstance(version, str) else version
+            tqdm_dict["v_num"] = version
+
+        return tqdm_dict
```

Comment:
for now, this could be a private utility function which takes the Lightning Module & Trainer as inputs and replicates the existing functionality. then we could remove all the `self.trainer` references here

The code flow will be very convoluted in the meantime but I don't see a better alternative. and in v1.7 we can break this chain and have the metrics properly go from lightning module to trainer  (via self.log) to progress bar (via the properties) so the progress bar can handle all the filtering & display here

Reply:
why is it bad to call self.trainer here?

Reply:
also how can the func be private if we are going to call it in properties.py? please see my other q's in https://github.com/PyTorchLightning/pytorch-lightning/pull/8985#issuecomment-901657773 :)